notificationservice/notification.go

```diff
@@ -0,0 +1,423 @@
+}
+
+func initSession(context *context.Context) *session.Session {
+	tokenError := mysession.UpdateSessionToken(context.Config.DefaultProfile, context.Config.DefaultRegion, context.Config.DefaultDurationForMFA, context)
+	if tokenError != nil {
+		context.Logger.Error(tokenError.Error())
+	}
+	currentSession, createSessionError := mysession.CreateSession(context, context.Config.DefaultProfile, &context.Config.DefaultRegion)
+	if createSessionError != nil {
+		context.Logger.Error(createSessionError.Error())
+	}
+	return currentSession
+}
+
+func (conn *Connection) verifyRemoteSinkConfigured() (snsTopicExists bool, sqsQueueExists bool, err error) {
+	snsTopicExists, err = conn.getSnsTopicAttributes()
+	if err != nil {
+		conn.context.Logger.Error("Error getting sns topic configuration: " + err.Error())
+	}
+	sqsQueueExists, err = conn.getSqsQueueAttributes()
+	if err != nil {
+		conn.context.Logger.Error("Error getting sqs queue configuration: " + err.Error())
+	}
+	return
+}
+
+type Message struct {
+	Type             string
+	MessageId        string
+	TopicArn         string
+	Subject          string
+	Message          string
+	Timestamp        string
+	SignatureVersion string
+	Signature        string
+	SigningCertURL   string
+	UnsubscribeURL   string
+}
+
+// Monitor queue, that delivers messages sent by cloud formation stack progress
+func (conn *Connection) MonitorQueue() {
+	waitTimeSeconds := int64(3)
+	receiveMessageInput := sqs.ReceiveMessageInput{
+		QueueUrl:        conn.sqsQueueOutput.QueueUrl,
+		WaitTimeSeconds: &waitTimeSeconds,
+	}
+
+	pw, table := initTableWriter()
+
+	tolerance, err := time.ParseDuration("1s")
+	if err != nil {
+		conn.context.Logger.Error(err.Error())
+	}
+	startReadingMessagesTime := time.Now().Add(-tolerance)
+
+	receivedAllMessages := true
+	for receivedAllMessages {
+		receivedMessages, err := conn.SqsClient.ReceiveMessage(&receiveMessageInput)
+		if err != nil {
+			println("Error reading messages", err.Error())
+		}
+		for e := range receivedMessages.Messages {
+			v := Message{}
+			jsonBlob := []byte(*receivedMessages.Messages[e].Body)
+			err = json.Unmarshal(jsonBlob, &v)
+			if err != nil {
+				conn.context.Logger.Error("error reading json message" + err.Error())
+			}
+
+			//DELETE READ MESSAGE (to prevent reading the same message multiple times)
+			conn.SqsClient.DeleteMessage(&sqs.DeleteMessageInput{
+				QueueUrl:      conn.sqsQueueOutput.QueueUrl,
+				ReceiptHandle: receivedMessages.Messages[e].ReceiptHandle,
+			})
+
+			//Parse property message
+			splittedMessage := strings.FieldsFunc(v.Message, func(r rune) bool { return r == '\n' })
+			messageMap := map[string]string{}
+			for messageNum := range splittedMessage {
+				messages := strings.FieldsFunc(splittedMessage[messageNum], func(r rune) bool { return r == '=' })
+				messageMap[messages[0]] = messages[1]
+			}
+			// Parse timestamp of message
+			messageArrivedTime, err := time.Parse("2006-01-02T15:04:05.000Z", v.Timestamp)
```

Comment:
Is it possible to name that date and time format with some meaningful name as a constant?